core/sdk-core/src/main/java/software/amazon/awssdk/core/metrics/CoreMetric.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.core.metrics;
+
+import java.time.Duration;
+import software.amazon.awssdk.annotations.SdkPreviewApi;
+import software.amazon.awssdk.annotations.SdkPublicApi;
+import software.amazon.awssdk.core.retry.RetryPolicy;
+import software.amazon.awssdk.metrics.MetricCategory;
+import software.amazon.awssdk.metrics.MetricLevel;
+import software.amazon.awssdk.metrics.SdkMetric;
+
+/**
+ * <b>NOTE:</b> This is a Preview API and is subject to change so it should not be used in production.
+ */
+@SdkPreviewApi
+@SdkPublicApi
+public final class CoreMetric {
+    /**
+     * The unique ID for the service. This is present for all API call metrics.
+     */
+    public static final SdkMetric<String> SERVICE_ID =
+        metric("ServiceId", String.class, MetricLevel.ERROR);
+
+    /**
+     * The name of the service operation being invoked. This is present for all
+     * API call metrics.
+     */
+    public static final SdkMetric<String> OPERATION_NAME =
+        metric("OperationName", String.class, MetricLevel.ERROR);
+
+    /**
+     * True if the API call succeeded, false otherwise.
+     */
+    public static final SdkMetric<Boolean> API_CALL_SUCCESSFUL =
+        metric("ApiCallSuccessful", Boolean.class, MetricLevel.ERROR);
+
+    /**
+     * The number of retries that the SDK performed in the execution of the request. 0 implies that the request worked the first
+     * time, and no retries were attempted.
+     */
+    public static final SdkMetric<Integer> RETRY_COUNT =
+        metric("RetryCount", Integer.class, MetricLevel.ERROR);
+
+    /**
+     * The duration of the API call. This includes all call attempts made.
+     *
+     * <p>{@code API_CALL_DURATION = CREDENTIALS_FETCH_DURATION + MARSHALLING_DURATION + SUM_ALL(BACKOFF_DELAY_DURATION) +
```

Comment:
Maybe we should do ~= since this function disregards small, constant time factors.